client/client.go

```diff
@@ -0,0 +1,112 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"gitlab.com/NebulousLabs/errors"
+)
+
+type (
+	// Error is the error type returned by the API in case the status code
+	// is not a 2xx code.
+	Error struct {
+		Message string `json:"message"`
+	}
+
+	// Client is a helper library for interacting with an API.
+	Client struct {
+		staticAddr string
+	}
+)
+
+// Error implements the error interface for the Error type. It returns only the
+// Message field.
+func (err Error) Error() string {
+	return err.Message
+}
+
+// NewClient creates a new Client for an API listening on the given address.
+func NewClient(addr string) *Client {
+	return &Client{
+		staticAddr: addr,
+	}
+}
+
+// readAPIError decodes and returns an api.Error.
+func readAPIError(r io.Reader) error {
+	var apiErr Error
+	b, _ := ioutil.ReadAll(r)
```

Comment:
Is this error safe to ignore?

Reply:
decoding will fail if this fails. Tbh this is also just a copy paste from accounts and skyd.